jans-cedarling/cedarling/src/authz/mod.rs

```diff
@@ -87,16 +84,12 @@ impl Authz {
     }
 
     // decode JWT tokens to structs AccessTokenData, IdTokenData, UserInfoTokenData using jwt service
-    pub (crate) fn decode_tokens<'a>(&'a self, request: &'a Request) -> Result<ProcessTokensResult<'a>, AuthorizeError> {
-            // decode JWT tokens to structs AccessTokenData, IdTokenData, UserInfoTokenData using jwt service
-        Ok(self
-            .config
-            .jwt_service
-            .process_tokens::<AccessTokenData, IdTokenData, UserInfoTokenData>(
-                &request.access_token,
-                &request.id_token,
-                Some(&request.userinfo_token),
-        )?)
+    pub (crate) fn decode_tokens<'a>(&'a self, request: &'a Request) -> Result<DecodedTokens<'a>, AuthorizeError> {
+        let access_token = request.access_token.as_ref().map(|tkn| self.config.jwt_service.process_token(TokenStr::Access(tkn))).transpose()?;
+        let id_token = request.id_token.as_ref().map(|tkn| self.config.jwt_service.process_token(TokenStr::Id(tkn))).transpose()?;
+        let userinfo_token = request.userinfo_token.as_ref().map(|tkn| self.config.jwt_service.process_token(TokenStr::Userinfo(tkn))).transpose()?;
+
+        Ok(DecodedTokens { access_token, id_token, userinfo_token })
```

Comment:
Each returned token is enum of all tokens.... it has no sense....

Reply:
refactored here [18ed767](https://github.com/JanssenProject/jans/pull/10436/commits/18ed7673bc485b13e045d2159b04de12cfc8641a).